### ISSUE TITLE
[FEATURE] Publishing onto F-Droid

### ISSUE THREAD

**Issue by slashdevslashurandom** [agrahn/Android-Password-Store#74]
### Is your feature request related to a problem? Please describe.

It seems like the original version of this app was also available on F-Droid for easy installation and updating, but is no longer available there.

### Describe the solution you'd like

It is worth considering submitting this fork to F-Droid for building. As all apps on F-Droid are built from source with trusted keys, this will also prevent situations where a malicious `.apk` file is submitted as if it were a release without the ability to anyone to audit the exact source code from which it's built.

### Describe alternatives you've considered

Alternatively, IzzyOnDroid could be an easier option, as it accepts releases in binary format -- although that option would not provide the security benefit listed above.

**Comment by agrahn** [comment]
Thanks a lot for your interest!

I think that having this app hosted on F-Droid would be very useful. On the other hand, I am afraid that this would considerably increase the number of user requests and issue reports that I would need to address. This might quickly go past my personal resources of time and maybe knowledge. Also, the original author might have objections against it. So, for now, I think I am not yet ready for publishing the app on F-Droid, but I will definitely keep it on the agenda (and leave this issue open).

**Comment by pony-montana** [comment]
Hi, this app works very well and is more stable than 99% of apps that people use regularly. The work-logic and the code quality are pretty good, with no obsolete and unnecessary depencies...
I think would be great to make a guthub release, and then publish it somewhere like fdroid, or I suggest https://zapstore.dev so people can discover it.

This app is a bit techie to use, require to understand pgp and how git works... So probably it will not attract a lot of noobie users that will overwhelm your issue-reports, but will most probably attract power users and potential new contributors that can start a positive feedback-loop contributing with bug reports and pull requests.

I hope you will take this path, I think you are making a great work keeping the repo alive after the previous mantainer quitted, and would be fantastic to spread this fantastic app to other people :)

**Comment by rdoeffinger** [comment]
I am not familiar with the code of this app and have limited time.
However I am maintaining a few Android apps like QuickDic for example and I would like this to be available again.
I would have considered creating my own fork, updating the code and publishing it, but since you are already doing the work that seems a bit of a waste of time.
If there is a way to get it published again I would appreciate it and am willing to contribute in whatever way would make it possible (I probably should get familiar with the code first though, one way I've done in the past was to convert to Kotlin or updating outdated APIs, if there is that kind of think that needs doing/would be interesting to do - happy to look at any tasks you have on your TODO list).

**Comment by agrahn** [comment]
> I am not familiar with the code of this app and have limited time. However I am maintaining a few Android apps like QuickDic for example and I would like this to be available again. I would have considered creating my own fork, updating the code and publishing it, but since you are already doing the work that seems a bit of a waste of time. If there is a way to get it published again I would appreciate it and am willing to contribute in whatever way would make it possible (I probably should get familiar with the code first though, one way I've done in the past was to convert to Kotlin or updating outdated APIs, if there is that kind of think that needs doing/would be interesting to do - happy to look at any tasks you have on your TODO list).

Thank you for your interest and offering help! I may definitely need support when it comes to larger code renovation tasks, e. g. API changes and such, since it is my first Kotlin+Android project. I have already fixed some issues that I found most bothering. Now, I plan to publish a GitHub release soon and to get familiar with the F-Droid-related work. May I get back to you via this channel, in case I need help? Also, feel free to submit an issue or a PR if you fixed something.

**Comment by rdoeffinger** [comment]
Yes, of course, feel free to reach out here. We can also find other communications channels if it should be necessary/helpful (my name is not very common, so you can e.g. find my email easily enough if needed, even though I prefer to not put it down here).
I managed to compile the app btw, but did not get any further than that so far.

**Comment by agrahn** [comment]
OK, it is done, APS is now available on F-Droid. Since the APK is signed by F-Droid, you will need to first uninstall the GitHub-released version from the phone before installing the F-Droid-built package. Alternatively, you can continue using the pre-release version from GitHub to keep up with the latest fixes and additions.
